src/Uno.Toolkit.RuntimeTests/Tests/ItemsRepeaterExtensionTests.ISupportIncrementalLoading.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Uno.Toolkit.RuntimeTests.Helpers;
+using Uno.Toolkit.UI;
+using Uno.UI.RuntimeTests;
+using Windows.Foundation;
+using System.Collections.ObjectModel;
+using System.Runtime.InteropServices.WindowsRuntime;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Controls;
+#else
+using ItemsRepeater = Microsoft.UI.Xaml.Controls.ItemsRepeater;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Controls;
+#endif
+
+namespace Uno.Toolkit.RuntimeTests.Tests;
+
+partial class ItemsRepeaterExtensionTests
+{
+	[TestMethod]
+	[RequiresFullWindow]
+	[RunsOnUIThread]
+	[DataRow(Orientation.Vertical)]
+	[DataRow(Orientation.Horizontal)]
+	public async Task When_Incremental_Load(Orientation orientation)
+	{
+		const int BatchSize = 25;
+
+		var source = new InfiniteSource<int>(async start =>
+		{
+			await Task.Delay(25);
+			return Enumerable.Range(start, BatchSize).ToArray();
+		});
+
+		var SUT = new Grid() { Height = 210, Width = 210, VerticalAlignment = VerticalAlignment.Bottom };
+		var ir = SetupIncrementalLoadingItemsRepeater(source, orientation);
+		var sv = SetupScrollViewer(ir);
+		SUT.Children.Add(sv);
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(SUT);
+		await Task.Delay(1000);
+		var initial = GetCurrenState();
+
+		(double? hOffset, double? vOffset) = orientation switch
+		{
+			Orientation.Vertical => ((double?)null, 10_000),
+			Orientation.Horizontal => (10_000, default),
+			_ => throw new NotSupportedException()
+		};
+
+		// scroll to bottom
+		sv.ChangeView(hOffset, vOffset, null, disableAnimation: true);
+		await Task.Delay(500);
+		await UnitTestsUIContentHelper.WaitForIdle();
+		var firstScroll = GetCurrenState();
+
+		// scroll to bottom
+		sv.ChangeView(hOffset, vOffset, null, disableAnimation: true);
+		await Task.Delay(500);
+		await UnitTestsUIContentHelper.WaitForIdle();
+		var secondScroll = GetCurrenState();
+
+		Assert.AreEqual(BatchSize * 1, initial.LastLoaded, "Should start with first batch loaded.");
+		Assert.AreEqual(BatchSize * 2, firstScroll.LastLoaded, "Should have 2 batches loaded after first scroll.");
+		Assert.IsTrue(initial.LastMaterialized < firstScroll.LastMaterialized, "No extra item materialized after first scroll.");
+		Assert.AreEqual(BatchSize * 3, secondScroll.LastLoaded, "Should have 3 batches loaded after second scroll.");
+		Assert.IsTrue(firstScroll.LastMaterialized < secondScroll.LastMaterialized, "No extra item materialized after second scroll.");
+
+		(int LastLoaded, int LastMaterialized) GetCurrenState() =>
+		(
+			source.LastIndex,
+			Enumerable.Range(0, source.LastIndex).Reverse().FirstOrDefault(x => ir.TryGetElement(x) != null)
+		);
+	}
+
+	private static ScrollViewer SetupScrollViewer(ItemsRepeater ir)
```

Comment:
try to group the helper methods together, by moving this above/under the ir one.